rxbindings/src/main/java/com/amplifyframework/rx/RxAuthBinding.java

```diff
@@ -110,9 +114,48 @@ public Single<AuthSignInResult> signIn(
             delegate.signIn(username, password, options, onResult, onError));
     }
 
+    @Override
+    public Single<AuthSignInResult> signInWithMagicLink(@NonNull String username,
+                                                        @NonNull AuthPasswordlessFlow flow,
+                                                        @NonNull String redirectURL,
+                                                        @NonNull AuthMagicLinkOptions options) {
+        return toSingle((onResult, onError) ->
+                delegate.signInWithMagicLink(
+                        username, flow, redirectURL, options, onResult, onError)
+        );
+    }
+
+    @Override
+    public Single<AuthSignInResult> confirmSignInWithMagicLink(@NonNull String challengeResponse,
+                                                               @NonNull
+                                                               AuthConfirmSignInOptions options) {
+        return toSingle((onResult, onError) ->
+                delegate.confirmSignInWithMagicLink(challengeResponse, options, onResult, onError));
+    }
+
+    @Override
+    public Single<AuthSignInResult> signInWithOTP(@NonNull String username,
+                                                  @NonNull AuthPasswordlessFlow flow,
+                                                  @NonNull AuthPasswordlessDeliveryDestination
+                                                              destination,
+                                                  @NonNull AuthOTPOptions options) {
+        return toSingle((onResult, onError) ->
+                delegate.signInWithOTP(username, flow, destination, options, onResult, onError));
+    }
+
+    @Override
+    public Single<AuthSignInResult> confirmSignInWithOTP(@NonNull String challengeResponse,
+                                                         @NonNull AuthConfirmSignInOptions
+                                                                 options) {
+        return toSingle((onResult, onError) ->
+                delegate.confirmSignInWithOTP(challengeResponse, options, onResult, onError));
+    }
+
     @Override
     public Single<AuthSignInResult> signIn(@Nullable String username, @Nullable String password) {
-        return toSingle((onResult, onError) -> delegate.signIn(username, password, onResult, onError));
+        return toSingle((onResult, onError) -> delegate.signIn(
```

Comment:
nit: maybe the formatting looks different from above APIs?

Reply:
yeah I just did that to avoid lint errors.